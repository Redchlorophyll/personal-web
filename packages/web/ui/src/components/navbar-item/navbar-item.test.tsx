import { render, screen } from '@testing-library/react';
import { NavbarItem } from './navbar-item';
import userEvent from '@testing-library/user-event';

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

const push = jest.fn();

const useRouter = jest.spyOn(require('next/router'), 'useRouter');

describe('<NavbarItem />', () => {
  useRouter.mockImplementation(() => ({
    route: '/',
    pathname: '',
    query: '',
    asPath: '/',
    push,
  }));

  test('should render properly', () => {
    const { baseElement } = render(
      <NavbarItem wip href="/">
        Home
      </NavbarItem>
    );

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });

  test('it should be able to click"', async () => {
    const user = userEvent.setup();
    render(<NavbarItem href="/about">about</NavbarItem>);

    const target = screen.getByRole('checkbox');

    await user.click(target);
    expect(push).toHaveBeenCalledWith('/about');
  });

  test('it should not push to other page if it has same page"', async () => {
    const user = userEvent.setup();
    render(<NavbarItem href="/">about</NavbarItem>);

    const target = screen.getByRole('checkbox');

    await user.click(target);
    expect(push).not.toHaveBeenCalledWith('/');
  });
});
