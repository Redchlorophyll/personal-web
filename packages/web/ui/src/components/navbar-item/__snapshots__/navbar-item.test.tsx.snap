// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavbarItem /> should render properly 1`] = `
<body>
  <div>
    <div
      class="relative h-[calc(100%+20px)]"
    >
      <input
        class="cursor-pointer w-full h-full peer absolute -translate-y-3 z-[2] opacity-0"
        type="checkbox"
      />
      <div
        class="absolute transition-all peer-hover:h-[65%] bg-red-200 h-0 w-full rounded-b-[3px] peer-checked:text-black-100 peer-checked:peer-hover:h-full peer-checked:h-full peer-checked:bg-primary-700"
      />
       
      <div
        class="relative p-[18px] transition-colors peer-hover:text-black-100 peer-checked:text-black-100 text-lg"
      >
        <p
          class="bg-red-700 absolute z-[1] py-[1px] px-[3px] text-xs -translate-y-1 right-0 text-black-100"
        >
          WIP
        </p>
        Home
      </div>
    </div>
  </div>
</body>
`;
